packages/bot-handler-avs/src/AVSHandler.ts

```diff
@@ -0,0 +1,178 @@
+import {CallingContent} from '@wireapp/core/dist/conversation/content';
+import {CallConfigData} from '@wireapp/api-client/dist/account';
+
+export class AVSHandler extends MessageHandler {
+  private wCall?: Wcall;
+  private wUser?: number;
+  private activeCalls: Record<string, Call> = {};
+  private readonly mediaStream: MediaStream = new MediaStream();
+
+  async handleEvent(payload: PayloadBundle): Promise<void> {}
+
+  async init(): Promise<void> {
+    const {clientId, userId} = this.account!;
+    const callingInstance = await getAvsInstance();
+    const wCall = this.configureCallingApi(callingInstance);
+    const wUser = this.createWUser(wCall, userId, clientId);
+    this.wUser = wUser;
+    this.wCall = wCall;
+  }
+
+  onIncomingCallMessage(callMessage: CallMessage): void {
+    const callContent: CallingContent = callMessage.content;
+
+    if (!callMessage.fromClientId) {
+      throw new Error('callMessage.fromClientId not found');
+    }
+
+    const avsResult = this.wCall!.recvMsg(
+      this.wUser!,
+      callContent,
+      callContent.length,
+      new Date().getSeconds(),
+      new Date(callMessage.timestamp).getSeconds(),
+      callMessage.conversation,
+      callMessage.from,
+      callMessage.fromClientId,
+    );
+
+    if (avsResult !== 0) {
+      throw new Error(`recvMsg failed with code "${avsResult}"`);
+    }
+
+    this.answerCall(callMessage);
+  }
+
+  startCall(conversationId: string, conversationType: CONV_TYPE, callType: CALL_TYPE): void {
+    this.wCall!.start(this.wUser!, conversationId, callType, conversationType, 0);
+  }
+
+  private answerCall(call: CallMessage) {
+    this.wCall!.answer(this.wUser!, call.conversation, CALL_TYPE.NORMAL, 0);
+  }
+
+  private configureCallingApi(wCall: Wcall): Wcall {
+    wCall.setLogHandler((level: LOG_LEVEL, message: string) => {
+      console.info(`${level}: ${message}`);
+    });
+
+    const avsEnv = AVS_ENV.DEFAULT;
+    wCall.init(avsEnv);
+    wCall.setUserMediaHandler(this.mediaHandler);
+    wCall.setMediaStreamHandler(() => {});
+    setInterval(() => wCall.poll(), 500);
+    return wCall;
+  }
+
+  private readonly mediaHandler = async (): Promise<MediaStream> => {
+    return this.mediaStream;
+  };
+
+  private readonly createWUser = (wCall: Wcall, selfUserId: string, selfClientId: string): number => {
+    /* cspell:disable */
+    const wUser = wCall.create(
+      selfUserId,
+      selfClientId,
+      () => {},
+      this.onSendCallMessage,
+      this.sendSFTRequest,
+      this.onIncomingCall,
+      () => {},
+      () => {},
+      () => {},
+      this.callClosed,
+      this.callConfigRequestHandler,
+      () => 0,
+      () => {},
+      () => {},
+    );
+
+    wCall.setParticipantChangedHandler(wUser, () => {});
+    return wUser;
+  };
+
+  private callConfigRequestHandler(convid: string, metrics_json: string, arg: number) {
+    this.account!.service!.account.getCallConfig()
+      .then((config: CallConfigData) => {
```

Comment:
Declaration possibly not needed.
```suggestion
      .then(config => {
```